libs/wingc/src/type_check.rs

```diff
@@ -811,6 +823,7 @@ impl<'a> TypeChecker<'a> {
 
 				let element_type = match *container_type {
 					Type::Array(t) => t,
+					Type::MutArray(t) => t,
```

Comment:
I wonder if we want to support literals defined at mutable. Maybe all container literals should be of the immutable variant, while allowing them to be assigned to a mutable variant:
```wing
let a: MutArray<num> =  // Here we're defining a mutable container
  [1,2,3];  // And assigning an immutable value to it during initialization.
```
This can also be left for the future and we can just "let" mutable literals exist for now...

Reply:
I think we would want both options.

This should work:

```js
let x = MutArray<num>[1,2,3];
```

And also this:

```js
let x: MutArray<num> = [1,2,3];
```

I've created #1117 to cover the second case.